infra/src/main/kotlin/fi/fta/geoviite/infra/map/AlignmentGeometry.kt

```diff
@@ -18,20 +26,46 @@ enum class MapAlignmentType {
     REFERENCE_LINE,
 }
 
-data class AlignmentHeader<T, U>(
-    val id: DomainId<T>,
-    val version: LayoutRowVersion<T>?,
-    val trackNumberId: DomainId<TrackLayoutTrackNumber>?,
+sealed class AlignmentHeader<T, U> {
+    abstract val id: DomainId<T>
+    abstract val trackNumberId: DomainId<TrackLayoutTrackNumber>?
+    abstract val name: AlignmentName
+    abstract val state: U
+    abstract val alignmentSource: MapAlignmentSource
+    abstract val alignmentType: MapAlignmentType
+    abstract val length: Double
+    abstract val boundingBox: BoundingBox?
+    abstract val segmentCount: Int
+}
+
+data class GeometryAlignmentHeader<T, U>(
+    override val id: DomainId<T>,
+    override val trackNumberId: DomainId<TrackLayoutTrackNumber>?,
+    override val name: AlignmentName,
+    override val state: U,
+    override val alignmentType: MapAlignmentType,
+    override val length: Double,
+    override val boundingBox: BoundingBox?,
+    override val segmentCount: Int,
+) : AlignmentHeader<T, U>() {
+    override val alignmentSource = MapAlignmentSource.GEOMETRY
+}
+
+data class LayoutAlignmentHeader<T : LayoutAsset<T>, U>(
+    override val id: IntId<T>,
+    val version: LayoutRowVersion<T>,
+    override val trackNumberId: IntId<TrackLayoutTrackNumber>?,
     val duplicateOf: IntId<LocationTrack>?,
-    val name: AlignmentName,
-    val state: U,
-    val alignmentSource: MapAlignmentSource,
-    val alignmentType: MapAlignmentType,
+    override val name: AlignmentName,
+    override val state: U,
```

Comment:
🤔 Oon jotenkin ite tottunut käyttämään pidempiä nimityksiä myös genericseille, erityisesti jos niitä on useampi käytössä (tässä siis voisi yhtälailla olla `State` geneerisen tyypin nimityksenä, joskin sekään ei välttämättä ole erityisen kuvaava). 

Tosin toinen koulukunta taas hyödyntää näitä tyypillisiä lyhyitä genericsejä helpomman bongauksen vuoksi, eikä niinkään ylläpidä geneerisen muuttujan nimessä kontekstitietoa siitä, minkälaista tyyppiä geneerinen muuttuja kuvaa.

Reply:
Samaa mieltä, kyllä tuo parantaisi tässä luettavuutta jos noilla olisi kunnon nimet. Tuo T on aika monessa paikkaa vain T joten hiukan houkuttaisi jättää se mutta jos nuo molemmat pidentäisi niin tämä voisi olla vaikkapa `<AssetType : LayoutAsset<AssetType>, StateType>`. Ei tuo ehkä liian pitkä ole -> muutan.

Reply:
Itse asiassa, nimeän nämä tuohon base classiin mutta miksipä ne ei olisi kiinnitettyjä lopullisissa luokissa. Voi poistaa vielä pari turhaa nullabilityä kun erottaa reference linen ja location trackin toisistaan ja geometryn tapauksessahan noihin on vain yksi mielekäs tyyppi mutenkin.